infra/modules/s3-encrypted/main.tf

```
@@ -0,0 +1,299 @@
+data "aws_caller_identity" "current" {}
+data "aws_region" "current" {}
+
+resource "aws_s3_bucket" "s3_encrypted" {
+  bucket = var.s3_bucket_name
+
+  # checkov:skip=CKV_AWS_144:Cross region replication not required by default
+  # checkov:skip=CKV2_AWS_61:Lifecycle policy will be added in later ticket for post-pilot cleanup
+  # checkov:skip=CKV2_AWS_62:Disable SNS requirement
+
+}
+
+resource "aws_s3_bucket_public_access_block" "s3_encrypted" {
+  bucket = aws_s3_bucket.s3_encrypted.id
+
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
+
+resource "aws_s3_bucket_policy" "s3_encrypted" {
+  bucket = aws_s3_bucket.s3_encrypted.id
+  policy = data.aws_iam_policy_document.s3_encrypted.json
+}
+
+resource "aws_s3_bucket_policy" "s3_encrypted_require_tls" {
+  bucket = aws_s3_bucket.s3_encrypted.id
+  policy = data.aws_iam_policy_document.require_tls.json
+}
+
+resource "aws_iam_role_policy" "read_s3" {
+  for_each = toset(var.read_role_names)
+  name     = "ReadAndDecryptS3"
```

Something like this:

```suggestion
  name     = "ReadAndDecryptS3-${var.environment}"
```